test/utils/tokenize.ts

```diff
@@ -482,6 +482,12 @@ export namespace Token {
             export const GreaterThanOrEqual = createToken('>=', 'keyword.operator.relational.cs');
         }
 
+        export namespace Word {
```

Comment:
Would `Pattern` work better here? Is it time to move `is` from being a normal keyword to being an operator?

Reply:
> Is it time to move is from being a normal keyword to being an operator?

I guess it makes sense. Both `as` and `is` are more of an operator compared to other keywords.

I was also wondering how specific we want to define the scopes. Whether we prefer

```
keyword.operator.word.as.cs
keyword.operator.word.is.cs
keyword.operator.word.and.cs
keyword.operator.word.or.cs
keyword.operator.word.not.cs
```

or we want more details

```
keyword.operator.word.cast.as.cs
keyword.operator.word.pattern.is.cs
keyword.operator.word.pattern.combinator.and.cs
keyword.operator.word.pattern.combinator.or.cs
keyword.operator.word.pattern.combinator.not.cs
```

Reply:
Personally, I like the more specific set. Does `word` add anything here though?

Reply:
Yes, because `keyword.operator` is for symbols and is coloured grey/white in most themes.

`keyword.operator.word` is chosen for word operators (for example, in python) and in most themes is coloured the same as normal keywords (blue in many themes).

Reply:
That makes sense. I think we should go with the specific set.

Reply:
@JoeRobich I noticed that `keyword.operator.word` is not set in the default themes of vscode:

```json
"scope": [
  "keyword.operator.new",
  "keyword.operator.expression",
  "keyword.operator.cast",
  "keyword.operator.sizeof",
  "keyword.operator.alignof",
  "keyword.operator.typeid",
  "keyword.operator.alignas",
  "keyword.operator.instanceof",
  "keyword.operator.logical.python",
  "keyword.operator.wordlike"
],
```

What would be a better fit for the pattern keywords then? C#'s old "is-type operator" would've been okay to be assigned to `keyword.operator.instanceof`, but now the `is` keyword strictly only takes patterns (type is merged into a type pattern) - and then there are also 3 extra keywords `and or not`.

If we just use `keyword.operator.pattern`, then people might complain that their `is` keyword is now suddenly not working (highlighted grey).

Reply:
Looking at other grammars, it seems that cpp emits both `keyword.operator.wordlike` then a specific operator scope like `keyword.operator.pattern.is`

https://github.com/microsoft/vscode/blob/b4cf1fa6029f3b05413a7c28b89971288a7b9f08/extensions/cpp/syntaxes/cpp.tmLanguage.json#L16465-L16472

Reply:
Personally I don't like that solution. It's unnecessary to have two scopes assigned when they could just be merged and have the same customisation levels. For example, `keyword.operator.wordlike.pattern.is` achieves the same. The only downside is that it's kinda too long?

Currently, the options I have in mind are:
* `keyword.operator.wordlike.pattern.*` &mdash; current approach, just `word` &rarr; `wordlike`
* `keyword.pattern.*` &mdash; shorter approach, would be treated as a `keyword` as themes won't have this scope yet
* `keyword.operator.pattern.*` &mdash; would be best if we could have this, but themes don't have this set yet and will appear grey (operator) and would appear to users as if highlighting doesn't work

Reply:
I like the first option because it works today and aligns with what other grammars are doing.